logic/src/commonMain/kotlin/com/wire/kalium/logic/feature/asset/GetPublicAssetUseCase.kt

```diff
@@ -0,0 +1,24 @@
+package com.wire.kalium.logic.feature.asset
+
+import com.wire.kalium.logic.data.asset.AssetRepository
+import com.wire.kalium.logic.data.user.UserAssetId
+import com.wire.kalium.logic.functional.suspending
+
+interface GetPublicAssetUseCase {
+    suspend operator fun invoke(assetKey: UserAssetId): PublicAssetResult
+}
+
+internal class GetPublicAssetUseCaseImpl(private val assetDataSource: AssetRepository) : GetPublicAssetUseCase {
+    override suspend fun invoke(assetKey: UserAssetId): PublicAssetResult = suspending {
+        assetDataSource.downloadPublicAsset(assetKey).fold({
+            PublicAssetResult.Failure
+        }) {
+            PublicAssetResult.Success(it)
+        }
+    }
+}
+
+sealed class PublicAssetResult {
+    class Success(val asset: ByteArray) : PublicAssetResult()
+    object Failure : PublicAssetResult()
```

Comment:
Should we maybe include the core failure as a value here?
```suggestion
    class Failure(val coreFailure: CoreFailure) : PublicAssetResult()
```

Reply:
Thanks @gongracr ! Applied the suggestion, and indeed we could need it ie, user feedback purposes 👍